src/Stack/Types/Config.hs

```diff
@@ -441,6 +443,21 @@ instance FromJSON ApplyGhcOptions where
             "everything" -> pure AGOEverything
             _ -> fail $ "Invalid ApplyGhcOptions: " ++ show t
 
+data AllowNewer
+  = AllowNewerAll
+  | AllowNewerNone
+  | AllowNewerPackages [String]
```

Comment:
I am wondering if it would be better if the type of the final data constructor was `![PackageName]` (ie `Distribution.Types.PackageName.PackageName` from the `Cabal` package), with consequent changes elsewhere. The `PackageName` type is used throughout Stack's code base.

Reply:
Fixed, thank you